warehouse/forklift/legacy.py

```diff
@@ -1322,6 +1336,30 @@ def file_upload(request):
                         "platform tag '{plat}'.".format(filename=filename, plat=plat),
                     )
 
+            try:
+                wheel_metadata_contents = _extract_wheel_metadata(temporary_filename)
+            except KeyError:
```

Comment:
@pradyunsg @dstufft how often do you think this would be hit? I don't want to be _too_ strict but think its best if we guard against this.

Reply:
With a well formed wheel file it should *never* happen.